packages/core/src/provider/validation.ts

```diff
@@ -0,0 +1,53 @@
+import { Address as zAddress } from "abitype/zod";
+import z from "zod";
+import { getChain } from "../utils/index.js";
+
+export const SmartAccountProviderOptsSchema = z.object({
+  /**
+   * The maximum number of times to try fetching a transaction receipt before giving up (default: 5)
+   */
+  txMaxRetries: z.number().optional(),
+
+  /**
+   * The interval in milliseconds to wait between retries while waiting for tx receipts (default: 2_000n)
+   */
+  txRetryIntervalMs: z.number().optional(),
+
+  /**
+   * The mulitplier on interval length to wait between retries while waiting for tx receipts (default: 1.5)
+   */
+  txRetryMulitplier: z.number().optional(),
+
+  /**
+   * used when computing the fees for a user operation (default: 100_000_000n)
+   */
+  minPriorityFeePerBid: z.bigint().optional(),
+});
+
+export const SmartAccountProviderConfigSchema = z.object({
+  rpcProvider: z.union([
+    z.string(),
+    z.any().refine((provider) => {
```

Comment:
we should type the `refinement` so we get accurate types here
```suggestion
    z.any().refine<PublicErc4337Client>((provider) => {
```